lib/rubocop/cop/rspec/memoized_helpers_in_example_group.rb

```diff
@@ -0,0 +1,136 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks if example groups contain too many `let` and `subject` calls.
+      #
+      # This cop is configurable using the `Max` option and the `AllowSubject`
+      # which will configure the cop to only register offenses on calls to
+      # `let` and not calls to `subject`.
+      #
+      # @example
+      #   # bad
+      #   describe MyClass do
+      #     let(:foo) { [] }
+      #     let(:bar) { [] }
+      #     let!(:booger) { [] }
+      #     subject { {} }
+      #     subject(:wat) { {} }
+      #     subject!(:boo) { {} }
+      #   end
+      #
+      #   describe MyClass do
+      #     let(:foo) { [] }
+      #     let(:bar) { [] }
+      #
+      #     context 'when stuff' do
+      #       let!(:booger) { [] }
+      #       subject { {} }
+      #       subject(:wat) { {} }
+      #       subject!(:boo) { {} }
+      #     end
+      #   end
+      #
+      #   # good
+      #   describe MyClass do
+      #     let(:foo) { [] }
+      #     let!(:booger) { [] }
+      #     subject { {} }
+      #     subject(:wat) { {} }
+      #     subject!(:boo) { {} }
+      #   end
+      #
+      #   describe MyClass do
+      #     context 'when stuff' do
+      #       let(:foo) { [] }
+      #       let(:bar) { [] }
+      #       let!(:booger) { [] }
+      #     end
+      #
+      #     context 'when other stuff' do
+      #       subject { {} }
+      #       subject(:wat) { {} }
+      #       subject!(:boo) { {} }
+      #     end
+      #   end
+      #
+      # @example with AllowSubject configuration
```

Comment:
👍
There's `MultipleSubjects` to take care of more than one subject.
However, unlike this cop, it has a deficiency of not counting inside nested example groups.